lib/active_model/serializer/adapter/json.rb

```diff
@@ -4,44 +4,122 @@ class ActiveModel::Serializer::Adapter::Json < ActiveModel::Serializer::Adapter
 
         def serializable_hash(options = nil)
           options ||= {}
+          hash = {}
+
           if serializer.respond_to?(:each)
-            result = serializer.map { |s| FlattenJson.new(s).serializable_hash(options) }
+            hash = serialize_array_without_root(serializer, options)
```

Comment:
What do you think about renaming this method to `serializable_hash_for_collection`, and extracting the else clause into a method called `serializable_hash_for_single_resource`? That's what I'd like to introduce in the JSONAPI adapter, so I'm up for discussion on that.